servicetalk-http-api/src/main/java/io/servicetalk/http/api/SingleAddressHttpClientBuilder.java

```diff
@@ -39,88 +49,255 @@
  * @param <R> the type of address after resolution (resolved address)
  */
 public abstract class SingleAddressHttpClientBuilder<U, R>
-        extends BaseSingleAddressHttpClientBuilder<U, R, ServiceDiscovererEvent<R>> {
-
-    @Override
-    public abstract SingleAddressHttpClientBuilder<U, R> ioExecutor(IoExecutor ioExecutor);
+        implements HttpClientBuilder<U, R, ServiceDiscovererEvent<R>> {
 
-    @Override
-    public abstract SingleAddressHttpClientBuilder<U, R> executionStrategy(HttpExecutionStrategy strategy);
-
-    @Override
-    public abstract SingleAddressHttpClientBuilder<U, R> bufferAllocator(BufferAllocator allocator);
-
-    @Override
+    /**
+     * Adds a {@link SocketOption} for all connections created by this builder.
+     *
+     * @param option the option to apply.
+     * @param value the value.
+     * @param <T> the type of the value.
+     * @return {@code this}.
+     * @see StandardSocketOptions
+     * @see ServiceTalkSocketOptions
+     */
     public abstract <T> SingleAddressHttpClientBuilder<U, R> socketOption(SocketOption<T> option, T value);
 
-    @Override
+    /**
+     * Enables wire-logging for connections created by this builder.
+     *
+     * @param loggerName The name of the logger to log wire events.
+     * @param logLevel The level to log at.
+     * @param logUserData {@code true} to include user data (e.g. data, headers, etc.). {@code false} to exclude user
+     * data and log only network events.
+     * @return {@code this}.
+     */
     public abstract SingleAddressHttpClientBuilder<U, R> enableWireLogging(String loggerName,
-                                                                           LogLevel logLevel,
-                                                                           BooleanSupplier logUserData);
+                                                                                LogLevel logLevel,
+                                                                                BooleanSupplier logUserData);
 
-    @Override
+    /**
+     * Configurations of various HTTP protocol versions.
+     * <p>
+     * <b>Note:</b> the order of specified protocols will reflect on priorities for
+     * <a href="https://tools.ietf.org/html/rfc7301">ALPN</a> in case the connections use TLS.
+     *
+     * @param protocols {@link HttpProtocolConfig} for each protocol that should be supported.
+     * @return {@code this}.
+     */
     public abstract SingleAddressHttpClientBuilder<U, R> protocols(HttpProtocolConfig... protocols);
```

Comment:
Consider adding `@see HttpProtocolConfigs` tag

Reply:
That would require the api module to depend on http-netty.